### ISSUE TITLE
[service-bus] Connection/network recovery

### ISSUE THREAD

**Issue by richardpark-msft** [Azure/azure-sdk-for-go#15170]

**Comment by richardpark-msft** [comment]
This task might be a bit large. Looking through the code base I see a few spots where retries are happening, but there are some bugs in the issue tracker (azure-service-bus-go) that show we seem to lose connections when there are network failures.

Adding this recoverability in JS was difficult, I expect this will be similar.

**Comment by richardpark-msft** [comment]
I'm taking a first stab at this for beta 1 (October release).

The big things:
- The processor type brings in built-in infinite retrying, which should be more durable than the previous APIs. 
- Our error classification has been improved - much thanks to the go-shuttle project (and @serbrech). I've added in some further checking and also allow for link and connection level recovery (we fallback to connection more than I'd like, but it's a first go).

As a first beta release bar I've got a stress test running, with network outages every minute for 5 seconds. This gives us some decent recovery testing, while also getting a good ballpark on our longevity.

There are a few spots that are being string-matched rather than doing a type/interface check. We need to work to eliminate these.
